app/src/main/java/org/eyeseetea/malariacare/SettingsActivity.java

```diff
@@ -63,11 +75,38 @@
      * shown on tablets.
      */
     private static final boolean ALWAYS_SIMPLE_PREFS = false;
+    private static final String TAG = ".Settings";
     private AutoCompleteEditTextPreference autoCompleteEditTextPreference;
+
+    /**
+     * Intent extra param that states that the login before changing critical info has been done
+     */
+    public static final String LOGIN_BEFORE_CHANGE_DONE="LOGIN_BEFORE_CHANGE_DONE";
+
+    /**
+     * Intent extra param that states that the login is being done due to an attempt to change the orgunit
+     */
+    public static final String SETTINGS_CHANGING_ORGUNIT="SETTINGS_CHANGING_ORGUNIT";
+
+    /**
+     * Intent extra param that states that the login is being done due to an attempt to change the server
+     */
+    public static final String SETTINGS_CHANGING_SERVER="SETTINGS_CHANGING_SERVER";
+
+
     protected void onCreate(Bundle savedInstanceState) {
+        Dhis2Application.bus.register(this);
         super.onCreate(savedInstanceState);
     }
 
+    @Override
+    public void onStop(){
+        try {
+            Dhis2Application.bus.unregister(this);
+        }catch(Exception e){}
```

Comment:
@adrianq the try/catch is not required, an exception should never be raised here.

A warning dialog might be a little bit confusing since the app will get back to the dashboard ( you're leaving the settingsactivity) and the potential (never expected) error is not related to you or anything you can do about it.